tensorflow/compiler/xla/service/gpu/target_util.cc

```diff
@@ -66,6 +71,30 @@ struct TargetIntrinsics GetIntrinsic(TargetIntrinsicID intrin) {
       return {llvm::Intrinsic::nvvm_barrier0,
               llvm::Intrinsic::amdgcn_s_barrier};
     }
+    case TargetIntrinsicID::kBlockDimx: {
+      return {llvm::Intrinsic::nvvm_read_ptx_sreg_ntid_x,
+              [](llvm::IRBuilder<>* b_) -> llvm::CallInst* {
+                return EmitDeviceFunctionCall("__ockl_get_local_size",
```

Comment:
Extract a constant?

Reply:
I'm not quite following you on this question.

This is to invoke:
https://github.com/RadeonOpenCompute/ROCm-Device-Libs/blob/master/ockl/inc/ockl.h#L438